### PR TITLE
build(deps): ts updates and linting

### PR DESCRIPTION
**Pull request by gkartalis**
This PR resolves [] 

### Description

- [x] commits linter auto fixes
- [x] updates some ts eslint deps due to this error that came up after [updating TS to v5](https://github.com/artsy/eigen/pull/8377)
```error
=============

WARNING: You are currently running a version of TypeScript which is not officially supported by @typescript-eslint/typescript-estree.

You may find that it works just fine, or you may not.

SUPPORTED TYPESCRIPT VERSIONS: >=3.3.1 <5.0.0

YOUR TYPESCRIPT VERSION: 5.0.2

Please only submit bug reports when using the officially supported version.

=============
```

### PR Checklist

- [ ] I have tested my changes on **iOS** and **Android**.
- [ ] I hid my changes behind a **[feature flag]**, or they don't need one.
- [ ] I have included **screenshots** or **videos**, or I have not changed the UI.
- [ ] I have added **tests**, or my changes don't require any.
- [ ] I added an **[app state migration]**, or my changes do not require one.
- [ ] I have documented any **follow-up work** that this PR will require, or it does not require any.
- [ ] I have added a **changelog entry** below, or my changes do not require one.

### To the reviewers 👀

- [ ] I would like **at least one** of the reviewers to **run** this PR on the simulator or device.

<details><summary>Changelog updates</summary>

#nochangelog

</details>

Need help with something? Have a look at our [docs], or get in touch with us.

[app state migration]: ../blob/main/docs/adding_state_migrations.md
[feature flag]: ../blob/main/docs/developing_a_feature.md
[docs]: ../blob/main/docs/README.md

### CODE REVIEW

File: src/app/Scenes/NewWorksFromGalleriesYouFollow/NewWorksFromGalleriesYouFollow.tsx [hunk 1]
```diff
@@ -11,16 +13,10 @@ import { screen } from "app/utils/track/helpers"
...
-import { NewWorksFromGalleriesYouFollow_artworksConnection$key } from "__generated__/NewWorksFromGalleriesYouFollow_artworksConnection.graphql"
 
 const SCREEN_TITLE = "New Works from Galleries You Follow"
 
-interface NewWorksFromGalleriesYouFollowScreenProps {}
-
-export const NewWorksFromGalleriesYouFollow: React.FC<
-  NewWorksFromGalleriesYouFollowScreenProps
-> = () => {
+export const NewWorksFromGalleriesYouFollow: React.FC<{}> = () => {
```
Comment (MrSltun) [inline src/app/Scenes/NewWorksFromGalleriesYouFollow/NewWorksFromGalleriesYouFollow.tsx:19]:
```suggestion
export const NewWorksFromGalleriesYouFollow: React.FC = () => {
```

File: src/app/Scenes/NewWorksFromGalleriesYouFollow/NewWorksFromGalleriesYouFollow.tsx [hunk 2]
```diff
@@ -88,12 +84,10 @@ const artworkConnectionFragment = graphql`
   }
 `
 
-export const NewWorksFromGalleriesYouFollowScreen: React.FC<
-  NewWorksFromGalleriesYouFollowScreenProps
-> = (props) => {
+export const NewWorksFromGalleriesYouFollowScreen: React.FC<{}> = () => {
```
Comment (MrSltun) [inline src/app/Scenes/NewWorksFromGalleriesYouFollow/NewWorksFromGalleriesYouFollow.tsx:87]:
```suggestion
export const NewWorksFromGalleriesYouFollowScreen: React.FC = () => {
```

File: src/app/Scenes/RecentlyViewed/RecentlyViewed.tsx [hunk 3]
```diff
@@ -11,14 +13,10 @@ import { screen } from "app/utils/track/helpers"
...
 import { graphql, useLazyLoadQuery, usePaginationFragment } from "react-relay"
-import { RecentlyViewedQuery } from "__generated__/RecentlyViewedQuery.graphql"
-import { RecentlyViewed_artworksConnection$key } from "__generated__/RecentlyViewed_artworksConnection.graphql"
 
 const SCREEN_TITLE = "Recently Viewed"
 
-interface RecentlyViewedScreenProps {}
-
-export const RecentlyViewed: React.FC<RecentlyViewedScreenProps> = () => {
+export const RecentlyViewed: React.FC<{}> = () => {
```
Comment (MrSltun) [inline src/app/Scenes/RecentlyViewed/RecentlyViewed.tsx:19]:
```suggestion
export const RecentlyViewed: React.FC = () => {
```

File: src/app/Scenes/RecentlyViewed/RecentlyViewed.tsx [hunk 4]
```diff
@@ -86,10 +84,10 @@ const artworkConnectionFragment = graphql`
   }
 `
 
-export const RecentlyViewedScreen: React.FC<RecentlyViewedScreenProps> = (props) => {
+export const RecentlyViewedScreen: React.FC<{}> = () => {
```
Comment (MrSltun) [inline src/app/Scenes/RecentlyViewed/RecentlyViewed.tsx:87]:
```suggestion
export const RecentlyViewedScreen: React.FC = () => {
```

**Review by MounirDhahri — DISMISSED** [review]

File: src/app/Scenes/SimilarToRecentlyViewed/SimilarToRecentlyViewed.tsx [hunk 5]
```diff
@@ -86,12 +84,10 @@ const artworkConnectionFragment = graphql`
   }
 `
 
-export const SimilarToRecentlyViewedScreen: React.FC<SimilarToRecentlyViewedScreenProps> = (
-  props
-) => {
+export const SimilarToRecentlyViewedScreen: React.FC<{}> = () => {
```
Comment (MrSltun) [inline src/app/Scenes/SimilarToRecentlyViewed/SimilarToRecentlyViewed.tsx:87]:
```suggestion
export const SimilarToRecentlyViewedScreen: React.FC = () => {
```

**Review by MrSltun — DISMISSED** [review]
Thank you George! ✨✨ 
I just added a few visual suggestions, but feel free to merge without them :)

**Review by olerichter00 — APPROVED** [review]
🌟

**Pull request merged**